### PR TITLE
fix allocations for snow and rain output

### PR DESCRIPTION
**Pull request by trontrytel**
# PULL REQUEST

This PR fixes some allocations related to outputting rain and snow. See the discussion here: https://github.com/CliMA/ClimaAtmos.jl/pull/816

@charleskawczynski - Is that what you had in mind?

## Purpose and Content
(Clearly and concisely state the purpose of this PR)

## Benefits and Risks
(State concisely the benefits to be derived from and the risks associated with this PR)

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- Fixes #
- Closes #

## PR Checklist
- [ ] This PR has a corresponding issue OR is linked to an SDI.
- [ ] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [ ] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [ ] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [ ] I linted my code on my local machine prior to submission OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code used in an integration test OR N/A.
- [ ] All tests ran successfully on my local machine OR N/A.
- [ ] All classes, modules, and function contain docstrings OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]

**Comment by charleskawczynski** [comment]
> @charleskawczynski - Is that what you had in mind?

Yes, exactly, thank you!

**Comment by charleskawczynski** [comment]
Sorry, @trontrytel, I just realized that `ᶜ3d_rain` is not unpacked

File: examples/hybrid/callbacks.jl [hunk 1]
```diff
@@ -203,8 +203,8 @@ function save_to_disk_func(integrator)
             )
 
         # rain vs snow
-        ᶜ3d_rain = @. ifelse(ᶜT >= FT(273.15), ᶜS_ρq_tot, FT(0))
-        ᶜ3d_snow = @. ifelse(ᶜT < FT(273.15), ᶜS_ρq_tot, FT(0))
+        @. ᶜ3d_rain = ifelse(ᶜT >= FT(273.15), ᶜS_ρq_tot, FT(0))
```
Comment (charleskawczynski) [inline examples/hybrid/callbacks.jl:206]: `ᶜ3d_rain` needs to be unpacked from `p` above

Comment (trontrytel) [reply examples/hybrid/callbacks.jl:206]: gotcha

Comment (trontrytel) [reply examples/hybrid/callbacks.jl:206]: Seems like we never use it in our CI. I'll switch it on for one case just to test it

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**